app/src/main/kotlin/com/mapbox/vision/teaser/BaseVisionFragment.kt

```diff
@@ -0,0 +1,39 @@
+package com.mapbox.vision.teaser
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import com.mapbox.vision.mobile.core.models.Country
+import com.mapbox.vision.teaser.utils.classification.SignResources
+import com.mapbox.vision.teaser.utils.dpToPx
+
+open class BaseVisionFragment : Fragment() {
```

Comment:
I'd just remembered an idea we've discussed yesterday - what if `BaseVisionFragment` extends `VisionEventsListener` ? Then every child will just override necessary method, and Activity won't need to call specific fragments at all - it'll just call `requireVisionFragment.onXXX` ?
@kiryldz @yuryybk wdyt?

Reply:
Generally, I agree with this, and it is OK to me.

The one opposite consideration that it might be confusing for people because VisionEventsListener should be registered in VisionManager, and we use it in BaseFragment only for code readability. But it just feeling

Reply:
I thinks it does not matter much, up to PR's author.

Reply:
Seems better to keep as is. It comes to empty methods in BaseFragment with unclear logic to override and the requirement to check and search in every fragment how it uses one or another callback